tests/test_raw_request.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import absolute_import, division, print_function
+
+import json
+import datetime
+from collections import OrderedDict
+
+import pytest
+
+import stripe
+from stripe.api_requestor import _json_encode_date_callback
+from stripe.api_version import _ApiVersion
+
+from tests.test_api_requestor import APIHeaderMatcher
+from tests.test_api_requestor import GMT1
+
+
+class TestRawRequest(object):
+    ENCODE_INPUTS = OrderedDict(
+        [
+            ("type", "standard"),
+            ("int", 123),
+            (
+                "datetime",
+                datetime.datetime(2013, 1, 1, second=1, tzinfo=GMT1()),
+            ),
+        ],
+    )
+    POST_REL_URL = "/v1/accounts"
+    GET_REL_URL = "/v1/accounts/acct_123"
+    POST_ABS_URL = stripe.api_base + POST_REL_URL
+    GET_ABS_URL = stripe.api_base + GET_REL_URL
+
+    @pytest.fixture(autouse=True)
+    def setup_stripe(self, http_client):
+        orig_attrs = {
+            "api_key": stripe.api_key,
+            "api_version": stripe.api_version,
+            "default_http_client": stripe.default_http_client,
+        }
+        stripe.api_key = "sk_test_123"
+        stripe.api_version = "2017-12-14"
+        stripe.default_http_client = http_client
+        yield
+        stripe.api_key = orig_attrs["api_key"]
+        stripe.api_version = orig_attrs["api_version"]
+        stripe.default_http_client = orig_attrs["default_http_client"]
+
+    def test_form_request_get(self, mock_response, check_call):
+        mock_response('{"id": "acct_123", "object": "account"}', 200)
+
+        resp = stripe.raw_request("get", self.GET_REL_URL)
+
+        check_call("get", abs_url=self.GET_ABS_URL)
+
+        deserialized = stripe.deserialize(resp)
+        assert isinstance(deserialized, stripe.Account)
+
+    def test_form_request_post(self, mock_response, check_call):
+        mock_response('{"id": "acct_123", "object": "account"}', 200)
+
+        expectation = "type=standard&int=123&datetime=1356994801"
+
+        resp = stripe.raw_request(
+            "post", self.POST_REL_URL, **self.ENCODE_INPUTS
+        )
+
+        check_call(
+            "post",
+            abs_url=self.POST_ABS_URL,
+            headers=APIHeaderMatcher(
+                content_type="application/x-www-form-urlencoded",
+                request_method="post",
+            ),
+            post_data=expectation,
+        )
+
+        deserialized = stripe.deserialize(resp)
+        assert isinstance(deserialized, stripe.Account)
+
+    def test_preview_request_post(self, mock_response, check_call):
+        mock_response('{"id": "acct_123", "object": "account"}', 200)
+
+        params = dict({"api_mode": "preview"}, **self.ENCODE_INPUTS)
+        expectation = json.dumps(
```

Comment:
A little stumped on how to hardcode this value without breaking Python 2.7. Might need to write a custom matcher that ignores ordering